### PR TITLE
Bugfix/move action capability consistent thread

### LINKED ISSUE #3691: FCL uses a thread-local cache, and MoveGroupMoveAction spawns endless threads.
### Description

To improve performance, FCL uses a thread-local cache of octomaps, `getShapeCache` in `collision_detection_fcl/src/collision_common.cpp`.

This would work well if the threads were consistent.

Unfortunately, the various move group capabilities all use the `MoveGroupMoveAction`, which calls `std::thread{ <callback> }.detach()` every time it is called.

This means that FCL is constantly re-populating empty `FCLShapeCache`s, wasting huge amounts of cpu time. 

### ROS Distro

Jazzy

### OS and version

Ubuntu 24.04

### Source or binary build?

Source

### If binary, which release version?

_No response_

### If source, which branch?

It looks like this is a problem on `rolling`, although I'm testing on a fork off `jazzy`.

### Which RMW are you using?

Cyclone

### Steps to Reproduce

 - Plan a motion using a move group capability, with a setup that uses FCL, on a robot with at least one attached collision object (I am using the pilz LIN planner, but that shouldn't matter).
 - Add a print like `RCLCPP_INFO_STREAM(getLogger(), "thread: " << std::this_thread::get_id() << " cache size: " << cache.map_.size())` to the `case shapes::MESH:` block of `collision_common.cpp`.
 - Observe the thread id changing, and the cache size going down to 0 a lot.

### Expected behavior

The FCLShapeCache should be preserved between multiple planning calls.

### Actual behavior

The FCLShapeCache is repeatedly destroyed, wasting time as it is rebuilt over and over.

### Backtrace or Console output

```
(gdb) break pthread_create
Breakpoint 1 at 0x74b0a3bb6bc8
(gdb) c
Continuing.
[Switching to Thread 0x74b03d7fa6c0 (LWP 945)]

Thread 44 "move_group" hit Breakpoint 1, 0x000074b0a3bb6bc8 in pthread_create
    () from /lib/x86_64-linux-gnu/libc.so.6
(gdb) bt
#0  0x000074b0a3bb6bc8 in pthread_create ()
   from /lib/x86_64-linux-gnu/libc.so.6
#1  0x000074b0a3e46eb1 in std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#2  0x000074b059b31223 in std::_Function_handler<void (std::shared_ptr<rclcpp_action::ServerGoalHandle<moveit_msgs::action::MoveGroup> >), move_group::MoveGroupMoveAction::initialize()::{lambda(std::shared_ptr<rclcpp_action::ServerGoalHandle<moveit_msgs::action::MoveGroup> > const&)#2}>::_M_invoke(std::_Any_data const&, std::shared_ptr<rclcpp_action::ServerGoalHandle<moveit_msgs::action::MoveGroup> >&&) () from /opt/ros/jazzy/lib/libmoveit_move_group_default_capabilities.so
#3  0x000074b059b41de3 in rclcpp_action::Server<moveit_msgs::action::MoveGroup>::call_goal_accepted_callback(std::shared_ptr<rcl_action_goal_handle_s>, std::array<unsigned char, 16ul>, std::shared_ptr<void>) ()
   from /opt/ros/jazzy/lib/libmoveit_move_group_default_capabilities.so
#4  0x000074b0a1db2f58 in rclcpp_action::ServerBase::execute_goal_request_received(int, action_msgs__msg__GoalInfo, rmw_request_id_s, std::shared_ptr<void>) ()
   from /opt/ros/jazzy/lib/librclcpp_action.so
#5  0x000074b0a1db406f in rclcpp_action::ServerBase::execute(std::shared_ptr<void> const&) () from /opt/ros/jazzy/lib/librclcpp_action.so
#6  0x000074b0a411e38b in rclcpp::Executor::execute_any_executable(rclcpp::AnyExecutable&) () from /opt/ros/jazzy/lib/librclcpp.so
#7  0x000074b0a4130d57 in rclcpp::executors::MultiThreadedExecutor::run(unsigned--Type <RET> for more, q to quit, c to continue without paging--
 long) () from /opt/ros/jazzy/lib/librclcpp.so
#8  0x000074b0a3e46db4 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x000074b0a3bb6aa4 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#10 0x000074b0a3c43a64 in clone () from /lib/x86_64-linux-gnu/libc.so.6
(gdb)
```

### PR DESCRIPTION
**Pull request by riv-mjohnson**
### Description

Uses a single persistent thread in the move action capability.
Should fix https://github.com/moveit/moveit2/issues/3691 for move action capabilities.

Update: now includes @riv-jkambites fix to make the persistent thread only use cpu when work is available.

When this is merged, could it be backported to Jazzy and Kilted?

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/moveit/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/moveit/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by riv-mjohnson** [comment]
The CI is failing with

```
    [test_get_cartesian_path_request_features_with_move_group-8] /home/runner/work/moveit2/moveit2/.work/target_ws/src/moveit2/moveit_ros/trajectory_cache/test/features/test_get_cartesian_path_request_features_with_move_group.cpp:105: Failure
    [test_get_cartesian_path_request_features_with_move_group-8] Expected equality of these values:
    [test_get_cartesian_path_request_features_with_move_group-8]   coll.queryList(exact_query).size()
    [test_get_cartesian_path_request_features_with_move_group-8]     Which is: 0
    [test_get_cartesian_path_request_features_with_move_group-8]   1
    [test_get_cartesian_path_request_features_with_move_group-8] Google Test trace:
    [test_get_cartesian_path_request_features_with_move_group-8] /home/runner/work/moveit2/moveit2/.work/target_ws/src/moveit2/moveit_ros/trajectory_cache/test/features/test_get_cartesian_path_request_features_with_move_group.cpp:90: CartesianStartStateJointStateFeatures
    [test_get_cartesian_path_request_features_with_move_group-8] 
    [test_get_cartesian_path_request_features_with_move_group-8] [  FAILED  ] MoveGroupFixture.GetCartesianPathRequestRoundTrip (4501 ms)
    [test_get_cartesian_path_request_features_with_move_group-8] [----------] 1 test from MoveGroupFixture (4501 ms total)
    [test_get_cartesian_path_request_features_with_move_group-8] 
    [test_get_cartesian_path_request_features_with_move_group-8] [----------] Global test environment tear-down
    [test_get_cartesian_path_request_features_with_move_group-8] [==========] 1 test from 1 test suite ran. (4502 ms total)
    [test_get_cartesian_path_request_features_with_move_group-8] [  PASSED  ] 0 tests.
    [test_get_cartesian_path_request_features_with_move_group-8] [  FAILED  ] 1 test, listed below:
    [test_get_cartesian_path_request_features_with_move_group-8] [  FAILED  ] MoveGroupFixture.GetCartesianPathRequestRoundTrip
```

Is this related to the changes made in this PR, or something which also fails in rolling?
